src/toast/tests/ops_sim_tod_catalog.py

```diff
@@ -0,0 +1,159 @@
+            catalog["transient_source"] = {
+                "ra_deg": 43,
+                "dec_deg": -43,
+                "freqs_ghz": [1.0, 1000.0],
+                "flux_density_Jy": [
+                    [10.0, 1.0],
+                    [30.0, 10.0],
+                ],
+                "times_mjd": [
+                    58849.0,
+                    58850.0,
+                ],
+            }
+
+            with open(self.catalog_file, "w") as f:
+                f.write(toml.dumps(catalog))
+
+        if self.comm is not None:
+            self.comm.barrier()
+
+        return
+
+    def test_sim_catalog(self):
+        # Create a fake ground data set for testing.  It targets a small patch at
+        # RA = [40, 44], Dec = [-44, -40]
+        data = create_ground_data(self.comm, turnarounds_invalid=True, hwp_rpm=0)
+
+        # Create an uncorrelated noise model from focalplane detector properties
+        default_model = ops.DefaultNoiseModel(noise_model="noise_model")
+        default_model.apply(data)
+
+        # Create some detector pointing matrices
+
+        nside = 1024
+
+        detpointing = ops.PointingDetectorSimple()
+        pixels = ops.PixelsHealpix(
+            nside=nside,
+            create_dist="pixel_dist",
+            detector_pointing=detpointing,
+        )
+        weights = ops.StokesWeights(
+            mode="IQU",
+            detector_pointing=detpointing,
+            hwp_angle=defaults.hwp_angle,
+        )
+
+        # Simulate point sources
+
+        sim_catalog = ops.SimCatalog(
+            catalog_file=self.catalog_file,
+            detector_pointing=detpointing,
+        )
+        sim_catalog.apply(data)
+
+        # Map the signal to check
+
+        binner = ops.BinMap(
+            pixel_pointing=pixels,
+            stokes_weights=weights,
+            noise_model=default_model.noise_model,
+        )
+
+        mapper = ops.MapMaker(
+            name="sources",
+            binning=binner,
+            template_matrix=None,
+            write_hits=True,
+            write_map=True,
+            write_cov=True,
+            write_invcov=True,
+            write_rcond=True,
+            keep_final_products=True,
+            output_dir=self.outdir,
+            map_rcond_threshold=1e-2,
+        )
+        mapper.apply(data)
+
+        if data.comm.world_rank == 0:
+            set_matplotlib_backend()
+            import matplotlib.pyplot as plt
+
+            fname_map = os.path.join(self.outdir, "sources_map.fits")
+            m = hp.read_map(fname_map, None)
+
+            nrow, ncol = 1, 3
+            fig = plt.figure(figsize=[6 * ncol, 4 * nrow])
+            for i, mm in enumerate(np.atleast_2d(m)):
+                hp.gnomview(
+                    mm, sub=[nrow, ncol, 1 + i], rot=[42, -42], xsize=800, reso=0.5
+                )
+
+            outfile = os.path.join(self.outdir, "map.png")
+            fig.savefig(outfile)
```

Comment:
Having a visual check is great.  If the sources were bright enough we might even be able to look at the relevant pixel closest to the source and confirm the map value is close to expected.